worlds/noita/__init__.py

```diff
@@ -0,0 +1,109 @@
+from .Locations import NoitaLocation, item_pickups
+
+from BaseClasses import Region, RegionType, Entrance, Item, ItemClassification, MultiWorld, Tutorial
+from .Options import noita_options
+from worlds.AutoWorld import World, WebWorld
+
+client_version = 1
+
+
+class NoitaWeb(WebWorld):
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Noita integration for Archipelago multiworld games.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["DaftBrit"]
+    )]
+    theme = "partyTime"
+
+
+class NoitaWorld(World):
+    """
+    Noita is a magical action roguelite set in a world where every pixel is physically simulated. Fight, explore, melt,
+    burn, freeze and evaporate your way through the procedurally generated world using spells you've created yourself.
+    """
+    game: str = "Noita"
+    option_definitions = noita_options
+    topology_present = False
+
+    item_name_to_id = item_table
+    location_name_to_id = item_pickups
+
+    data_version = 5
+    forced_auto_forfeit = True
+    web = NoitaWeb()
+
+    def generate_basic(self):
+
+        if self.world.bad_affects[self.player].value:
+            pool_option = 0
+        else:
+            pool_option = 1
+        junk_pool: Dict[str, int] = {}
+        junk_pool = item_pool_weights[pool_option].copy()
+        # Generate item pool
+        itempool: List = []
+
+        for i in range(1, 1 + self.world.total_locations[self.player].value):
+            itempool += self.world.random.choices(list(junk_pool.keys()), weights=list(junk_pool.values()))
+
+        # Convert itempool into real items
+        itempool = list(map(lambda name: self.create_item(name), itempool))
+
+        self.world.itempool += itempool
+
+    def create_regions(self) -> None:
+        menu = create_region(self.world, self.player, "Menu")
+        mines = create_region(self.world, self.player, "Mines",
+                                  [f"Chest{i + 1}" for i in range(self.world.total_locations[self.player].value)])
+
+        connection = Entrance(self.player, "Lobby", menu)
+        menu.exits.append(connection)
+        connection.connect(mines)
+
+        self.world.regions += [menu, mines]
+
+        create_events(self.world, self.player)
+
+    def fill_slot_data(self):
+        return {
+            "seed": "".join(self.world.slot_seeds[self.player].choice(string.digits) for i in range(16)),
+            "totalLocations": self.world.total_locations[self.player].value,
+            "badAffects": self.world.bad_affects[self.player].value,
+        }
+
+    def create_item(self, name: str) -> Item:
+        item_id = item_table[name]
+        if name == "Heart":
+            classification = ItemClassification.progression
+        elif name.startswith("Wand"):
+            classification = ItemClassification.useful
+        else:
+            classification = ItemClassification.filler
```

Comment:
If the "bad" item has a negative effect on the player, it should be classified as a trap